src/spawn/arguments.ts

```diff
@@ -13,6 +13,13 @@ export type CliVerbOptions = {
   cliVerb: string;
 };
 
+export interface ArgumentMappings {
+  [key: string]: Mapping;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Mapping = string | ((val: any) => string[]);
```

Comment:
I'd prefer not to turn off the lint warnings - perhaps this could be `unknown`?

Reply:
When we change to `unknown` here, we need annotate the mapping function argument type with unknown,
```ts
let argMapping: ArgumentMappings = {
    enablePending: (enablePending: unknown) => [enablePending ? '--enable-pending' : '--no-enable-pending']
}
```
or do a explicit cast:
```ts
let argMapping = {
    enablePending: (enablePending: boolean) => [enablePending ? '--enable-pending' : '--no-enable-pending']
} as ArgumentMappings;
```
Is that ok?

Reply:
Hm, I'd rather not do an explicit cast unless we have to, as (like `any`) it's turning off the type checking. Maybe we can bring in an implicit type parameter or something - I'll have a play.

Reply:
Ah, right - so the type checking is already pretty nobbled by this point, because of the way the rest of the mapper works (there are also already some explicit casts in the branches above). This is one of the reasons we rewrote the mapper for the native verifier in the master branch. 

I think `unknown` is probably the better way-  I'll push a commit that does that.

Reply:
(I prefer `unknown`, because really we don't know what the type will be at that point. If we did rewrite this mapper from scratch, we'd want to do it in a way that the type information was retained. I think this is an artefact of this originally being pure JS)